File: elixir/bob/bob.exs
defmodule Bob do
  def screaming(input) do
    String.match?(input, ~r/^[^a-z]*\p{Lu}+[^a-z]*!?$/) ->
  end

  def hey(input) do
    cond do
      String.ends_with?(input, "?") ->
        "Sure."
      String.trim(input) == "" ->
        "Fine. Be that way!"
      screaming(input) ->
        "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: haskell/bob/src/Bob.hs
module Bob (responseFor) where

import Data.Char

responseFor :: String -> String
responseFor xs
  | null trimmed = "Fine. Be that way!"
  | not (null alpha) && all isUpper alpha = "Whoa, chill out!"
  | last trimmed == '?' = "Sure."
  | otherwise = "Whatever."
  where trimmed = filter (not . isSpace) xs
        alpha = filter isAlpha xs
